Sources/EnumKit/Core/CaseAccessible.swift

```diff
@@ -18,8 +18,9 @@ public extension CaseAccessible {
     
     /// Check if an enum case matches another case
     func matches(case: Self) -> Bool {
-        let targetStr = `case`.label
-        return label == targetStr
+        var root = self
+        var `case` = `case`
+        return memcmp(&root, &`case`, MemoryLayout<Self>.size) == 0
```

Comment:
I'll back this as a robust solution for matching, considering this is the whole enum case and so should have the case index in memory.